spring-integration-test/src/main/java/org/springframework/integration/test/mock/MockMessageHandler.java

```diff
@@ -85,6 +87,7 @@ protected MockMessageHandler(ArgumentCaptor<Message<?>> messageArgumentCaptor) {
 	 * @param nextMessageConsumer the Consumer to handle the next incoming message.
 	 * @return this
 	 */
+	@SuppressWarnings("NullAway") // See github.com/uber/NullAway/issues/1075
```

Comment:
I'm not sure why we need this here.
That `Consumer<Message<?>>` cannot accept a null.
The message is never null in the integration flow.

Reply:
Oh! Yes, I see.
And yes, that function definitely can return null according to the `AbstractReplyProducingMessageHandler.handleRequestMessage()` contract.

Let's see if this helps:
```
protected Function<Message<?>, ? extends @Nullable Object> lastFunction;
```
And respectively, that `handleNextAndReply()`  contract.
Plus, probably `messageFunctions` signature.

Reply:
It looks really such an elegant approach !!
But still getting the same error.. Could be a limitation of the analyzer.

I apply your type signature improvement and keep the `@SuppressWarnings`.

BTW, the comment referencing issue looks like this is the exact same issue we're hitting.